### ISSUE TITLE
QUIC libressl Interop sometimes fails

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#2688]
### Detailed Description of the Problem

it is not 100% reproducable, I'd say it fails about 1 time out of 10...

this time it failed on C1 test:
https://github.com/haproxy/haproxy/actions/runs/10462824762

logs be kept for a week:
https://github.com/haproxy/haproxy/actions/runs/10462824762/artifacts/1830128133

### Expected Behavior

to pass every time

### Steps to Reproduce the Behavior

run QUIC Interop suite on HAproxy built with LibreSSL

### What is your configuration?

```haproxy
default QUIC Interop
```

### Output of `haproxy -vv`

```plain
no ((
```

**Comment by a-denoyelle** [comment]
`C1` causes random loss, and sometimes it may be cause the test to be really difficult or near impossible to test. As such, it also fails frequently with various different combinations of client/server on https://interop.seemann.io/. What is important for this test is that it passes most of the times :)

**Comment by chipitsine** [comment]
it looks more robust on aws-lc.
if we believe those failures are expected, shall we exclude that from weekly CI ?

**Comment by a-denoyelle** [comment]
I'm not sure. @haproxyFred has worked on this test to stabilize it, maybe he has an opinion on this ?

**Comment by haproxyFred** [comment]
C1/L1 are not supposed to always pass. By nature they randomly fail. What is suspect is when they **always** fail against a client. This is the case for the aws-lc interop runner where C1 or L1, I do not remember, always fails with s2n-client.

**Comment by chipitsine** [comment]
In theory if we would have been reporting test result like that
https://docs.gitlab.com/ee/ci/testing/unit_test_report_examples.html, we
could compare result between runs. Currently it does not look useful

**Comment by chipitsine** [comment]
one more question.

do you monitor weekly builds faulire ?
I do not want to pollute the issue tracker if you already examine failures.

if issues are better, I can try to setup automatic issue in case of failure

**Comment by haproxyFred** [comment]
I am not sure to understand what you mean by build failures. I guess you include in these failures the interop tests failures.

About these interop tests, we check every day only the official interop test runner as mentioned by Amaury here: https://interop.seemann.io/ About the CI, I do not know at all how it works with the interop runner. As the interop tests are run between haproxy and several clients, some tests failures may be due to the client. These failures must lead to haproxy gh issues reported here only in these cases:

- when they are not reported as alreay failed by the official interop test runner. As we check them, we decide if the is a problem or not,
- about C1/L1, when they very often fail (with more than 60% of the clients)

I have noticed that you run the network performance tests (Measurements results). You should not run these tests with the CI. They are very very few chances that some performance issues may be related to the TLS stack tested by the different interop tests runners coupled with the CI.

In conclusion, I think this issue must be closed.

**Comment by chipitsine** [comment]
by build failures I mean generally any failure of weekly run LibreSSL/AWS-LC QUIC Interop, including build and test failures.

ok, what if we'll do it like that...

current pipelines will be split into 2:

1) all testcases without C1/L1. if such build execution fails, it will create github issue
2) C1/L1 only



as for network measurements, I didn't find a way to turn them off. but I'll try again

**Comment by haproxyFred** [comment]
> 1. all testcases without C1/L1. if such build execution fails, it will create github issue

Does this mean that the CI would **automatically** create a GH issue for each "haproxy red light" here: https://interop.seemann.io ? This is a bad idea.

Aww I had just a look to the CI actions for haproxy. I see that you test only two clients picoquic and quic-go. That said there would  be useless newly created GH  issues already reported by the official interop runner.

So, I propose you to have a look at the CI reports for the AWS-LC and Libressl actions weekly. So, please do not automatically create GH issues when they fail. And I would add ngtcp2 to the list of the clients.

One more thing, I see you build the interop runner one time for each client. Why not building only one time the interop runner and then run the tests for each client?

A last question: when an CI action fails for the interop runner tests, where can I retrieve the interop runner logs?

**Comment by chipitsine** [comment]
> > 1. all testcases without C1/L1. if such build execution fails, it will create github issue
> 
> Does this mean that the CI would **automatically** create a GH issue for each "haproxy red light" here: https://interop.seemann.io ? This is a bad idea.

no, I do not mean https://interop.seemann.io

I mean weekly build set up in haproxy repo:

https://github.com/haproxy/haproxy/actions/workflows/quic-interop-aws-lc.yml
https://github.com/haproxy/haproxy/actions/workflows/quic-interop-libressl.yml

for those weekly builds we can try to create GH issues (if you do not monitor those builds)



> 
> Aww I had just a look to the CI actions for haproxy. I see that you test only two clients picoquic and quic-go. That said there would be useless newly created GH issues already reported by the official interop runner.

official Interop runner runs against QuicTLS.
ok, if you think it is useless, even better, I can leave as is 

> 
> So, I propose you to have a look at the CI reports for the AWS-LC and Libressl actions weekly. So, please do not automatically create GH issues when they fail. And I would add ngtcp2 to the list of the clients.

I'll try ngtcp2. last time it failed all tests on LibreSSL: https://github.com/haproxy/haproxy/issues/2599
I investigated, from my point of view there's something wrong on the LibreSSL side
but at least I can try aws-lc, why not

> 
> One more thing, I see you build the interop runner one time for each client. Why not building only one time the interop runner and then run the tests for each client?

it is built once for LibreSSL and once for AWS-LC. all tests run on the same image.
but I'll check
> 
> A last question: when an CI action fails for the interop runner tests, where can I retrieve the interop runner logs?


"goodput"  and "crosstraffic" logs are huge, up to 600Mb.
that's why I set up retention for 6 days:


<img width="925" alt="image" src="https://github.com/user-attachments/assets/5f6f3c9f-5fe4-4ca1-801a-da8af1a0d989">

**Comment by haproxyFred** [comment]
> I'll try ngtcp2. last time it failed all tests on LibreSSL: #2599

This is not a problem if you do not create automatically create GH issues.

> I investigated, from my point of view there's something wrong on the LibreSSL side
> but at least I can try aws-lc, why not

Ok

**Comment by haproxyFred** [comment]
> it is built once for LibreSSL and once for AWS-LC. all tests run on the same image.
> but I'll check

Have a look to the -c option of the run.py interop  runner python script.

**Comment by haproxyFred** [comment]
> "goodput" and "crosstraffic" logs are huge, up to 600Mb.

These are network performance tests which are useless to be run. We have the official interop runner test for those.

**Comment by chipitsine** [comment]
ok, I'll take mentioned considerations
